app/src/main/java/com/futurice/freesound/feature/common/ui/adapter/DiffUtilCallback.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.feature.common.ui.adapter;
+
+import com.futurice.freesound.feature.common.DisplayableItem;
+
+import android.support.annotation.NonNull;
+import android.support.v7.util.DiffUtil;
+
+import java.util.List;
+
+final class DiffUtilCallback extends DiffUtil.Callback {
+
+    @NonNull
+    private final List<DisplayableItem> oldItems;
+
+    @NonNull
+    private final List<DisplayableItem> newItems;
+
+    @NonNull
+    private final ItemComparator comparator;
+
+    DiffUtilCallback(@NonNull final List<DisplayableItem> oldItems,
+                     @NonNull final List<DisplayableItem> newItems,
+                     @NonNull final ItemComparator comparator) {
+        this.oldItems = oldItems;
```

Comment:
I'm probably contradicting myself here, but I think this should have null checks on the values. I say this because the values are calculated, rather than being simply and reliable created via Dagger.